primitives/trie/src/storage_proof.rs

```diff
@@ -32,27 +32,27 @@ use crate::LayoutV1 as Layout;
 /// the serialized nodes and performing the key lookups.
 #[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
 pub struct StorageProof {
-	trie_nodes: Vec<Vec<u8>>,
+	trie_nodes: BTreeSet<Vec<u8>>,
```

Comment:
So btreeset will produce always ordered proof, but it will also allow decoding proof with non ordered node.

Actually forcing order on decode will break existing code, so it is probably better this way.